easybuild/toolchains/compiler/intel_compilers.py

```diff
@@ -59,5 +61,18 @@ def _set_compiler_vars(self):
     def set_variables(self):
         """Set the variables."""
 
+        if self.options.get('oneapi', False):
+            self.COMPILER_CXX = 'icpx'
+            self.COMPILER_CC = 'icx'
+            self.COMPILER_F77 = 'ifx'
+            self.COMPILER_F90 = 'ifx'
+            self.COMPILER_FC = 'ifx'
+            # fp-model source is not supported by icx but is equivalent to precise
+            self.options.options_map['defaultprec'] = ['ftz', 'fp-speculation=safe', 'fp-model precise']
+            # icx doesn't like -fp-model fast=1; fp-model fast is equivalent
+            self.options.options_map['loose'] = ['fp-model fast']
```

Comment:
Does it handle -fp-model fast=2 then? I.e. veryloose

Reply:
So we should perhaps override veryloose too?

Reply:
It wouldn't make a difference in practice apart from a line in the log, unless at some point in the future fast=2 will be supported again.

I'm happy to override it but then we may need to add a version check in the future to re-enable it I suppose

Reply:
https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/compiler-options/floating-point-options/fp-model-fp.html

Reply:
"There is currently no difference between fast=1 and fast=2."

But for some reason fast=1 is flat out rejected with an error (so docs are wrong here), fast=2 as accepted with a warning

Reply:
done